tests/test-sources/plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,55 @@
+{
+  empty = {
+    plugins.schemastore.enable = true;
+  };
+
+  example = {
+    plugins = {
+      lsp = {
+        enable = true;
+
+        servers = {
+          jsonls.enable = true;
+          yamlls.enable = true;
+        };
+      };
+
+      schemastore = {
+        enable = true;
+
+        json = {
+          enable = true;
+          settings = {
+            replace."package.json" = {
+              description = "package.json overridden";
+              fileMatch = [ "package.json" ];
+              name = "package.json";
+              url = "https://example.com/package.json";
+            };
+            extra = [
+              {
+                description = "My custom JSON schema";
+                fileMatch = "foo.json";
+                name = "foo.json";
+                url = "https://example.com/schema/foo.json";
+              }
+              {
+                description = "My other custom JSON schema";
+                fileMatch = [
+                  "bar.json"
+                  ".baar.json"
+                ];
+                name = "bar.json";
+                url = "https://example.com/schema/bar.json";
+              }
+            ];
+          };
+        };
+        yaml = {
```

Comment:
Would be nice to have additional `withJson` and `withYaml` tests to verify it's still valid when only one is in use.

Reply:
E.g.

```nix
withJson = {
  plugins = {
    lsp = {
      enable = true;
      servers.jsonls.enable = true;
    };

    schemastore = {
      enable = true;

      json = {
        enable = true;
        settings = { };
      };
    };
  };
};

withYaml = {
  plugins = {
    lsp = {
      enable = true;
      servers.yamlls.enable = true;
    };

    schemastore = {
      enable = true;

      yaml = {
        enable = true;
        settings = { };
      };
    };
```

Reply:
True, done.